### ISSUE TITLE
CI: Add integration tests for drivers

### ISSUE THREAD

**Issue by straight-shoota** [crystal-lang/crystal-db#161]
It would be nice to have automated integration tests with at least the major driver implementations (the ones mentioned in the readme). This would help us identify issues that only show when actual drivers are used and validates that changes in this shard ensure compatibility with driver implementations.

Inspired by #160

**Comment by bcardiff** [comment]
Yes, but sometimes we do need breaking changes. So these should be smoke and optional

**Comment by straight-shoota** [comment]
I think it might even be a good practice to add compatibility in the drivers first before actually breaking anything. Maybe this won't work well for all cases, but it should work for many.

And we should probably consider some stability guarantees for `crystal-db` (i.e. a stable 1.0 release).
